openapi/social/social.yaml

```diff
@@ -142,6 +142,86 @@ paths:
           in: query
           name: dryRun
           description: 'If `true`, will simply return any errors this Message will get from posting. Will return an empty 200 if there are none.'
+    get:
+      summary: List Social Messages
+      operationId: get-messages
+      responses:
+        '200':
+          description: OK
+          content:
+            application/vnd.api+json:
+              schema:
+                type: object
+                properties:
+                  data:
+                    type: array
+                    items:
+                      $ref: '#/components/schemas/messages'
+                  links:
+                    type: object
+                    properties:
+                      self:
+                        type: string
+                        format: uri
+                      first:
+                        type: string
+                        format: uri
+                        description: Provides a link back to the first page of results.
+                      next:
+                        type: string
+                        format: uri
+                        description: The URI at which the next page of results can be gotten from.
+      description: |-
+        [Status](https://developers.vendasta.com/platform/ZG9jOjEwMTU2NTYy-versioning): `Proposed`
+        
+         An endpoint dedicated to listing all the posts made by the business filtered by the selected parameters.
+
+      parameters:
+        - schema:
+            type: string
+            example: AG-12345
+          in: query
+          name: 'filter[businessId]'
+          description: Returns all the posts for a business.
+        - schema:
+            type: string
+          in: query
+          name: 'page[cursor]'
+          description: The cursor stores all your filters and current location in the list to allow paging over the results in smaller batches. The value will be provided in the response links.
+        - schema:
+            type: integer
+            example: 20
+            default: 25
+          in: query
+          name: 'page[limit]'
+          description: The maximum number of Social Posts you would like to return.
+        - schema:
+            type: string
+            example: '2022-02-04T16:52:34.000Z'
+          in: query
+          name: 'filter[start]'
+          description: The starting time of the posts to be listed. Times used must be in ISO 8601 format, UTC time.
+        - schema:
+            type: string
+            example: '2022-02-04T16:52:34.000Z'
+          in: query
+          name: 'filter[end]'
+          description: The ending time of the posts to be listed. Times used must be in ISO 8601 format, UTC time.
```

Comment:
Which database field(s) is used in the where filter? We might want to communicate that so users are not guessing between scheduledAt, createdAt, postedAt.

We could copy the pattern from the [list purchases operation](https://developers.vendasta.com/platform/474f69c0b91b8-list-purchases) which uses `filter[createdAt][<]` and `filter[createdAt][>]` so they can add other fields in the future.

Reply:
while filtering, we use a field called `orderByPostedDateTime` present in [parent post table](https://console.cloud.google.com/datastore/databases/-default-/entities;kind=SocialPosts;ns=__$DEFAULT$__/query/kind;queryModel=1%7CWH%7C1%7C14%2FinternalPostId%7CEQ%7CSTR%7C44%2FSocialPosts-55c26aa7145a5e6e8247d6311b5af9c4?project=repcore-demo). This field is based on `postedDateTime` though.

Reply:
So for messages that have been scheduled but not yet posted does the system populate the orderByPostedDateTime field get populated with the scheduled date instead?

Reply:
It does with `postDateTime`. For example , [this post](https://console.cloud.google.com/datastore/databases/-default-/entities;kind=CuratedContentPost;ns=__$DEFAULT$__/query/kind;queryModel=1%7CWH%7C1%7C14%2FinternalPostId%7CEQ%7CSTR%7C88%2FCuratedContentPost-55c26aa7145a5e6e8247d6311b5af9c4-CCA-0453E4901E3645C9BFF2A5EA4233B907?project=repcore-demo) , this is not yet posted, but there is a `OrderByPostedDateTime` which is being populated to be same as the `postDateTime`.

Reply:
Since there is some smart logic using start and end instead of column names makes sense. We should rename them `filter[startAt]` and `filter[endAt]` to follow the convention of ending field names with At when they contain times.

It would be nice to describe the smart logic using terms that partners can understand instead of using the filter name in the description.

Reply:
I updated the filter names. If you can come up with a better description please add it.